SO-5596: snapshot import with unpublished references

cmark opened the pull request.

cmark committed:
refactor(test): rename method in SnomedImportApiTest

cmark committed:
chore(test): fix incorrect case in import api test method names

cmark committed:
fix(snomed): properly process Snapshot RF2 files with mixed...

...versioned and unversioned content by putting them into a single
snapshot slice instead of separating the unpublished components from the
snapshot versioned content. (a versioned description referencing an
unversioned concept is considered a valid snapshot content)

Resolves https://snowowl.atlassian.net/browse/SO-5596

AAAlinaaa commented on snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/io/SnomedImportApiTest.java:
```diff
@@ -653,10 +653,42 @@ public void import37NewOwlParentInSecondImportBatch() {
+	}
+  	
+  	@Test
+	public void import39SnapshotContentWithUnpublishedReferences() throws Exception {
```
Shouldn't this be delta? The name is the same as above only the number is different.

cmark replied:
Yes.

cmark replied:
Also, the assertion requires one extra line to ensure that the import completes but reports an error.

cmark commented:
7.x port PR available: #1159

AAAlinaaa reviewed (approved):
LGTM

cmark committed:
fix(test): give a better name to the newly introduced snapshot/delta...

...import test cases

Assert that in case of a delta export of such RF2 content the system
responds with an error message.

Relates to https://snowowl.atlassian.net/browse/SO-5596

nagyo reviewed (approved):
LGTM! 🍏
